be/src/io/cache_input_stream.cpp

```diff
@@ -415,18 +373,24 @@ StatusOr<std::string_view> CacheInputStream::peek(int64_t count) {
     SharedBufferPtr sb;
     ASSIGN_OR_RETURN(auto s, _sb_stream->peek_shared_buffer(count, &sb));
     if (_enable_populate_cache) {
-        _populate_cache_from_zero_copy_buffer(s.data(), _offset, count, sb);
+        _populate_to_cache(s.data(), _offset, count, sb);
     }
     return s;
 }
 
-void CacheInputStream::_populate_cache_from_zero_copy_buffer(const char* p, int64_t offset, int64_t count,
-                                                             const SharedBufferPtr& sb) {
+void CacheInputStream::_populate_to_cache(const char* p, int64_t offset, int64_t count, const SharedBufferPtr& sb) {
     int64_t begin = offset / _block_size * _block_size;
     int64_t end = std::min((offset + count + _block_size - 1) / _block_size * _block_size, _size);
     p -= (offset - begin);
     auto f = [sb, this](const char* buf, size_t off, size_t size) {
+        DCHECK(off % _block_size == 0);
         SCOPED_RAW_TIMER(&_stats.write_cache_ns);
+
```

Comment:
Move the scoped timer below after checking its existence.

Reply:
done